packages/block-editor/src/components/list-view/index.js

```diff
@@ -173,6 +183,65 @@ function ListView(
 		]
 	);
 
+	// If a selection is made outside the block list,
+	// for example, in the Block Editor,
+	// try to expand the block list tree.
+	useEffect( () => {
+		if (
+			! hasFocus &&
+			Array.isArray( selectedBlockParentClientIds ) &&
+			selectedBlockParentClientIds.length
+		) {
+			setExpandedState( {
+				type: 'expand',
+				clientIds: selectedBlockParentClientIds,
+			} );
+		}
+	}, [ hasFocus, selectedBlockParentClientIds ] );
+
+	useEffect( () => {
```

Comment:
This is just a WIP experiment to scroll to the selected element in the tree.

Here it is in action:

https://user-images.githubusercontent.com/6458278/148503872-a5f97694-f477-43be-b8a5-4c7c670f3989.mp4


The `hasFocus` trick is unreliable it seems. We need a way to isolate the select event to the block editor, so that the scroll and the above expand only occurs when a user clicks on a block in the editor.

Needs a lot more work, plus testing how it interacts with the expander hook above.